test/mocks/mocks.go

```diff
@@ -404,7 +405,7 @@ func ServiceBindingRequestMock(
 			ApplicationSelector: v1alpha1.ApplicationSelector{
 				Group:       "apps",
 				Version:     "v1",
-				Resource:    "deployments",
+				Resource:    applicationResourceKind,
```

Comment:
`applicationResourceKind` name is misleading here. Perhaps would be better to receive a `schema.GroupVersionResource` as argument, and use it to feed the `ApplicationSelector` field.

The other advantage of that is the GVR can be obtained through existing APIs, so no typos.

Reply:
@isutton I have addressed your comments. Please review this PR.